src/gui/newwizard/setupwizardcontroller.cpp

```diff
@@ -1,409 +1,156 @@
-            bool vfsModeIsExperimental = false;
-
-            switch (bestAvailableVfsMode()) {
-            case Vfs::WindowsCfApi:
-                vfsIsAvailable = true;
-                enableVfsByDefault = true;
-                vfsModeIsExperimental = false;
-                break;
-            case Vfs::WithSuffix:
-                vfsIsAvailable = true;
-                enableVfsByDefault = false;
-                vfsModeIsExperimental = true;
-                break;
-            default:
-                break;
-            }
-
-            _currentPage = new AccountConfiguredWizardPage(FolderMan::suggestSyncFolder(_accountBuilder.serverUrl(), _accountBuilder.displayName()), vfsIsAvailable, enableVfsByDefault, vfsModeIsExperimental);
-            _wizardWindow->displayPage(_currentPage, 2);
-        };
-
-        if (_accountBuilder.authType() == DetermineAuthTypeJob::AuthType::Basic) {
-            auto strategy = dynamic_cast<HttpBasicAuthenticationStrategy *>(_accountBuilder.authenticationStrategy());
-            Q_ASSERT(strategy != nullptr);
-
-            auto checkBasicAuthJob = Jobs::CheckBasicAuthJobFactory(_accessManager, strategy->username(), strategy->password(), this).startJob(_accountBuilder.serverUrl());
-
-            auto showCredentialsPageAgain = [this, checkBasicAuthJob](const QString &error) {
-                checkBasicAuthJob->deleteLater();
-
-                if (!error.isEmpty()) {
-                    _wizardWindow->showErrorMessage(error);
-                }
-
-                _currentPage = new BasicCredentialsSetupWizardPage(_accountBuilder.serverUrl());
-                _wizardWindow->displayPage(_currentPage, 1);
-            };
-
-            connect(checkBasicAuthJob, &CoreJob::finished, this, [moveToFinalPage, checkBasicAuthJob, showCredentialsPageAgain, this, strategy]() {
-                if (checkBasicAuthJob->success()) {
-                    if (checkBasicAuthJob->result().toBool()) {
-                        auto fetchUserInfoJob = FetchUserInfoJobFactory::fromBasicAuthCredentials(_accessManager, strategy->username(), strategy->password(), this).startJob(_accountBuilder.serverUrl());
-
-                        connect(fetchUserInfoJob, &CoreJob::finished, this, [this, strategy, moveToFinalPage, showCredentialsPageAgain, fetchUserInfoJob] {
-                            if (fetchUserInfoJob->success()) {
-                                auto result = fetchUserInfoJob->result().value<FetchUserInfoResult>();
-
-                                Q_ASSERT(result.userName() == strategy->username());
-
-                                _accountBuilder.setDisplayName(result.displayName());
-
-                                moveToFinalPage();
-                            } else {
-                                showCredentialsPageAgain(tr("Failed to fetch user display name"));
-                            }
-                        });
+    OC_ENFORCE(_currentState != nullptr);
```

Comment:
Im still pretty sure that is not needed in release builds.
You either handle errors or you assume they don't happen in release....